clients/da-vinci-client/src/test/java/com/linkedin/davinci/repository/NativeMetadataRepositoryTest.java

```diff
@@ -159,21 +180,71 @@ protected TestNMR(ClientConfig clientConfig, VeniceProperties backendConfig, Clo
     }
 
     @Override
-    protected StoreConfig getStoreConfigFromSystemStore(String storeName) {
+    protected StoreConfig fetchStoreConfigFromRemote(String storeName) {
       StoreConfig storeConfig = mock(StoreConfig.class);
       when(storeConfig.isDeleting()).thenReturn(false);
       return storeConfig;
     }
 
     @Override
-    protected Store getStoreFromSystemStore(String storeName, String clusterName) {
+    protected Store fetchStoreFromRemote(String storeName, String clusterName) {
       Store store = mock(Store.class);
       when(store.getName()).thenReturn(storeName);
       when(store.getReadQuotaInCU()).thenReturn(1L);
       return store;
     }
 
     @Override
+    protected SchemaData getSchemaData(String storeName) {
```

Comment:
With your refactor where we pushed a lot of the system store specific logic down to the `ThinClientMetaStoreBased` and the intent of TestNMR was to provide unit testing for common logic in `NativeMetadataRepository` I wonder if we need to provide such complicated and system store specific implementation of `getSchemaData`. Can just mock it with some in memory map for testing?